src/material/datepicker/datepicker.spec.ts

```diff
@@ -1191,9 +1191,9 @@ describe('MatDatepicker', () => {
             .not.toContain('ng-invalid');
       }));
 
-      it('should update validity when switching between null and invalid', fakeAsync(() => {
+      it('should update validity when switching between valid and invalid', fakeAsync(() => {
```

Comment:
I don't think this behavior change is correct, hopefully by addressing the comments above this will be fixed.